### PR TITLE
Use log probs for paraformer

### PR DESCRIPTION
**Pull request by pengzhendong**

### CODE REVIEW

File: sherpa-onnx/csrc/math.h [hunk 1]
```diff
@@ -103,5 +103,18 @@ std::vector<int32_t> TopkIndex(const T *vec, int32_t size, int32_t topk) {
   return index;
 }
 
+template <class T>
```
Comment (csukuangfj) [inline sherpa-onnx/csrc/math.h]: Please follow
https://github.com/k2-fsa/sherpa-onnx/blob/80060c276d2d5d75c9b2f24a132d7694ed1e4942/sherpa-onnx/csrc/offline-ctc-greedy-search-decoder.cc#L35-L39

to use `std::max_element` to replace `ArgMax()`.

You don't need to reimplement the wheel.

File: sherpa-onnx/csrc/offline-paraformer-greedy-search-decoder.cc [hunk 2]
```diff
@@ -6,26 +6,30 @@
...
+  std::vector<int64_t> shape = log_probs.GetTensorTypeAndShapeInfo().GetShape();
   int32_t batch_size = shape[0];
   int32_t num_tokens = shape[1];
+  int32_t vocab_size = shape[2];
 
   std::vector<OfflineParaformerDecoderResult> results(batch_size);
 
-  const int64_t *p = token_num.GetTensorData<int64_t>();
+  const float *p = log_probs.GetTensorData<float>();
   for (int32_t i = 0; i != batch_size; ++i) {
```
Comment (csukuangfj) [inline sherpa-onnx/csrc/offline-paraformer-greedy-search-decoder.cc]:
```suggestion
  
  for (int32_t i = 0; i != batch_size; ++i) {
    const float *p = log_probs.GetTensorData<float>() + i * num_tokens * vocab_size;
```

File: sherpa-onnx/csrc/math.h (hunk 1 above)
Comment (pengzhendong) [reply sherpa-onnx/csrc/math.h]: `p_log_probs` is `const float *`, why do we use `static_cast` again?

Comment (csukuangfj) [reply sherpa-onnx/csrc/math.h]: > `p_log_probs` is `const float *`, why do we use `static_cast` again?

yes, you are right. please remove the cast.

**Comment by csukuangfj** [comment]
Thanks!

**Pull request merged**